### ISSUE TITLE
Port CIFAR data augmentation to KeystoneML master

### ISSUE THREAD

**Issue by tomerk** [amplab/keystone#242]

**Comment by shivaram** [comment]
cc @Vaishaal - A draft of the code is at https://github.com/shivaram/keystone/blob/keystone-vw-experiments/src/main/scala/nodes/images/RandomFlips.scala 

Right now the code does both random patches and the center + corners, but we should separate them out into different nodes

**Comment by Vaishaal** [comment]
awesome,

**Comment by ericmjonas** [comment]
I know this is basically a function node that does a flatmap, but in the interest of code reuse, would it make sense to take in arbitrary Image->Image functions? I'd love to be able to quickly write a "rotate image slightly" function of Image -> Image type and then have this node just Do The Right Thing.

**Comment by shivaram** [comment]
Well I've already started refactoring to help some of this - I split the code into two parts one is a FunctionNode to just extract random patches and the second a RandomFlipper node (i.e. something that flips horizontally with some chance). We can refactor the second function to just be `RandomImageTranslator` that takes in Image => Image translate function and a probability. Would that meet your use case ?

Basically the `FunctionNode` is necessary to go from one Image to a bunch of patches. And then on for each patch we can have a number of transformers.

**Comment by etrain** [comment]
Added this to 0.3.0 because I think it's highly useful and we're pretty close to done.
